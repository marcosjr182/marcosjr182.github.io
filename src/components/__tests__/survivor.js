import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Survivor from '../survivor';

const survivor = {
  name: 'Test',
  age: '10',
  gender: 'M'
}

describe('Test suite for Survivor', () => {
  it('Suvivor should render info correctly', () => {
    const wrapper = shallow(<Survivor {...{ ...survivor, distance: '10' }} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  })

  it('should not break if it does not receive a distance', () => {
    const survivor = {...survivor, gender: 'F' }
    const wrapper = shallow(<Survivor {...survivor} />)
    expect(toJson(wrapper)).toMatchSnapshot();
  })
})
