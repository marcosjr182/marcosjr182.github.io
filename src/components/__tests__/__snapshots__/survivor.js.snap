exports[`Test suite for Survivor Suvivor should render info correctly 1`] = `
<div
  className="col-xs-12 survivor ">
  <div
    className="col-xs-12 name">
    Test
  </div>
  <div
    className="col-xs-12 info">
    MALE
     | 
    10
  </div>
  <Component
    columns="6" />
  <Distance />
</div>
`;
